resources/kprove.mak

```diff
@@ -36,28 +54,26 @@ ifdef DEBUG
 KPROVE_OPTS+=--debug-z3-queries --log-rules
 endif
 
+TIMEOUT?=30m
+
 #
 # Settings
 #
 
-# path to this file
-THIS_FILE:=$(abspath $(lastword $(MAKEFILE_LIST)))
-# path to root directory
-ROOT:=$(abspath $(dir $(THIS_FILE))/../)
-
-RESOURCES:=$(ROOT)/resources
-
-SPECS_DIR:=$(ROOT)/specs
-
-K_VERSION   :=$(shell cat $(BUILD_DIR)/.k.rev)
-KEVM_VERSION:=$(shell cat $(BUILD_DIR)/.kevm.rev)
+K_VERSION_FILE   :=$(BUILD_DIR)/.k.rev
+KEVM_VERSION_FILE:=$(BUILD_DIR)/.kevm.rev
+K_VERSION        :=$(shell cat $(K_VERSION_FILE))
+KEVM_VERSION     :=$(shell cat $(KEVM_VERSION_FILE))
 
 K_REPO_DIR:=$(abspath $(BUILD_DIR)/k)
 KEVM_REPO_DIR:=$(abspath $(BUILD_DIR)/evm-semantics)
 
 K_BIN:=$(abspath $(K_REPO_DIR)/k-distribution/target/release/k/bin)
 
-KPROVE:=$(K_BIN)/kprove -v --debug -d $(KEVM_REPO_DIR)/.build/defn/java -m VERIFICATION --z3-impl-timeout 500 \
+TIMEOUT_PREFIX:=timeout -s SIGINT -k 10s --foreground $(TIMEOUT)
+
+KPROVE:=$(TIMEOUT_PREFIX) $(K_BIN)/kprove -v --debug -d $(KEVM_REPO_DIR)/.build/defn/java -m VERIFICATION \
+        --z3-impl-timeout 500 --shutdown-wait-time 5000 \
```

Comment:
Is `--shutdown-wait-time 5000` a new option added to K?

Reply:
No, it's old. But it only makes sense in the context of SIGINT (or ctrl+C). That's why I added it now. When kprove gets SIGINT, it interrupts the main thread and waits for 5 seconds until main threads propagates the exception and prints all it wants. Whether this really happens depends on what kprove was doing at the time of interruption.